Analysis/Tasks/PWGHF/HFJpsiToEECandidateSelector.cxx

```diff
@@ -0,0 +1,226 @@
+//    mass  dcaxy dcaz  pt_e
+constexpr double cuts[npTBins][nCutVars] =
+  {{0.5, 0.2, 0.4, 1},  /* pt<0.5   */
+   {0.5, 0.2, 0.4, 1},  /* 0.5<pt<1 */
+   {0.5, 0.2, 0.4, 1},  /* 1<pt<2   */
+   {0.5, 0.2, 0.4, 1},  /* 2<pt<3   */
+   {0.5, 0.2, 0.4, 1},  /* 3<pt<4   */
+   {0.5, 0.2, 0.4, 1},  /* 4<pt<5   */
+   {0.5, 0.2, 0.4, 1},  /* 5<pt<7   */
+   {0.5, 0.2, 0.4, 1},  /* 7<pt<10  */
+   {0.5, 0.2, 0.4, 1}}; /* 10<pt<15 */
+
+/// Struct for applying Jpsi selection cuts
+
+struct HFJpsiToEECandidateSelector {
+
+  Produces<aod::HFSelJpsiToEECandidate> hfSelJpsiToEECandidate;
+
+  Configurable<double> d_pTCandMin{"d_pTCandMin", 0., "Lower bound of candidate pT"};
+  Configurable<double> d_pTCandMax{"d_pTCandMax", 50., "Upper bound of candidate pT"};
+
+  Configurable<double> d_pidTPCMinpT{"d_pidTPCMinpT", 0.15, "Lower bound of track pT for TPC PID"};
+  Configurable<double> d_pidTPCMaxpT{"d_pidTPCMaxpT", 10., "Upper bound of track pT for TPC PID"};
+
+  Configurable<double> d_TPCNClsFindablePIDCut{"d_TPCNClsFindablePIDCut", 70., "Lower bound of TPC findable clusters for good PID"};
+  Configurable<double> d_nSigmaTPC{"d_nSigmaTPC", 3., "Nsigma cut on TPC only"};
+
+  /// Gets corresponding pT bin from cut file array
+  /// \param candpT is the pT of the candidate
+  /// \return corresponding bin number of array
+  template <typename T>
+  int getpTBin(T candpT)
+  {
+    double pTBins[npTBins + 1] = {0, 0.5, 1., 2., 3., 4., 5., 7., 10., 15.};
+    if (candpT < pTBins[0] || candpT >= pTBins[npTBins]) {
+      return -1;
+    }
+    for (int i = 0; i < npTBins; i++) {
+      if (candpT < pTBins[i + 1]) {
+        return i;
+      }
+    }
+    return -1;
+  }
+
+  /// Selection on goodness of daughter tracks
+  /// \note should be applied at candidate selection
+  /// \param track is daughter track
+  /// \return true if track is good
+  template <typename T>
+  bool daughterSelection(const T& track)
+  {
+    if (track.charge() == 0) {
+      return false;
+    }
+    if (track.tpcNClsFound() == 0) {
+      return false; //is it clusters findable or found - need to check
+    }
+    return true;
+  }
+
+  /// Conjugate independent toplogical cuts
+  /// \param hfCandProng2 is candidate
+  /// \param trackPositron is the track with the positron hypothesis
+  /// \param trackElectron is the track with the electron hypothesis
+  /// \return true if candidate passes all cuts
+  template <typename T1, typename T2>
+  bool selectionTopol(const T1& hfCandProng2, const T2& trackPositron, const T2& trackElectron)
+  {
+    auto candpT = hfCandProng2.pt();
+    int pTBin = getpTBin(candpT);
+    if (pTBin == -1) {
+      return false;
+    }
+
+    if (candpT < d_pTCandMin || candpT >= d_pTCandMax) {
+      return false; //check that the candidate pT is within the analysis range
+    }
+
+    if (TMath::Abs(InvMassJpsi(hfCandProng2) - RecoDecay::getMassPDG(443)) > cuts[pTBin][0]) {
```

Comment:
shouldn't this bee InvMassJpsiToEE ?

Reply:
yes, I have found that. It will modified next commit.